### PR TITLE
Slice final: refactor and freestyle examples

### PR DESCRIPTION
**Pull request by rjar2020**
- Refactor. I felt bad about calling it collections in the aftermatch. Slices is more precise.
- One more kata with string slices, trying regex built-in capabilities.
- Additional insights: 
 a) Great handling of nil (null) scenarios by the standard lib, helps to reduce boilerplate, boosting readability
 b) In the tests, you can put multiple examples if you workout the output format.

### CODE REVIEW

File: slices/findd.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var digitRegexp = regexp.MustCompile("[0-9]+")
+
+//FindFirstConsecutiveDigits returns an slice containg the first sequence of integers in the input string slice
+func FindFirstConsecutiveDigits(sequence []string) []int {
```
Comment (rjar2020) [inline slices/findd.go:12]: This is a freestyle exercise, so a thoroughly code review is highly appreciated @JuanCalcagno416, thanks in advance!

**Review by JuanCalcagno416 — APPROVED** [review]
LGTM and liked the approach of doing something different having the base of the slices :smiling_imp:

File: slices/findd_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestFindFirstConsecutiveDigits(t *testing.T) {
+
+	matchDigits := func(t testing.TB, got, want []int) {
+		t.Helper()
+		if !reflect.DeepEqual(got, want) {
```
Comment (jsalinaspolo) [inline slices/findd_test.go:13]: I think you are gonna love https://github.com/stretchr/testify to get nice asserts

Comment (rjar2020) [reply slices/findd_test.go:13]: Javi, good stuff! I'm trying it already -> [PR for structs](https://github.com/rjar2020/learn-go-with-tests/pull/8)

A refactor is coming,  also gonna try [gorc](https://github.com/stretchr/gorc) <3

**Pull request merged**